### ISSUE TITLE
[Chore] Gitflow can be improved

### ISSUE THREAD

**Issue by longnd** [Vivek-sudo/nimble-assessment.backend#7]
Issue
---
While the 2 applications use pull requests to separate the work, there are room for improvement
- Some PRs handle multiple responsibilities, e.g. 
  + https://github.com/Vivek-sudo/nimble-assessment.server/pull/1 is responsible for scafolding the project setup, models (keyword, user), initializing the keyword service, etc
   + https://github.com/Vivek-sudo/nimble-assessment.server/pull/4
- There were usually giant commits, e.g. this PR https://github.com/Vivek-sudo/nimble-assessment.server/pull/1/commits has a single commit only. Having regular commits, each target an atomic change, would be helpful to understand each change easily.

Expected
--
- Proper [git-flow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) workflow usage.

**Comment by Vivek-sudo** [comment]
I completely agree to having regular commits and atomic changes. But as this project had to be made from scratch, I used to do major changes in one go. But once the project is setup with initial features and everything, I would work according to sprints where features will be picked up one by one and committed according to that.
